### PR TITLE
[FIX] hr_holidays_attendance: Public leave in _get_overtime_leave_domain

### PR DESCRIPTION
**Pull request by hwangh95**
The _get_overtime_leave_domain was added to help check for overtime on
employee leaves, but failed to account for public leaves

Note:
Backport of commit 1407052d3ff0663180ea11342569ace643c0cd00 from
odoo/odoo#141235

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by hwangh95** [comment]
@sofiagvaladze Sorry for the ping, but if you could quickly look at this that would be great. At the very least before the Christmas holidays, since there are people who might log hours on the public holiday.

**Review by sofiagvaladze — CHANGES_REQUESTED** [review]
@hwangh95 sorry for late reply, I was on vacation myself :innocent: 

I have just one question and then we can proceed :smile:

File: addons/hr_holidays_attendance/models/hr_attendance.py [hunk 1]
```diff
@@ -9,4 +9,5 @@ class HrAttendance(models.Model):
 
     def _get_overtime_leave_domain(self):
         domain = super()._get_overtime_leave_domain()
-        return AND([domain, [('holiday_id.holiday_status_id.time_type', '=', 'leave')]])
+        # resource_id = False => Public holidays
```
[the remark this answers is not in the document]
Comment (hwangh95) [reply addons/hr_holidays_attendance/models/hr_attendance.py:12]: I was pretty sure that the change did nothing in the end. resource_list already is initialized with a `self.env['resource.resource']` inside and an empty model.id is False

You can see it here
[1407052#diff-a39d1344f686aade27285d6860b04689032d63189ee42676318b21556140421fL502](https://github.com/odoo/odoo/commit/1407052d3ff0663180ea11342569ace643c0cd00#diff-a39d1344f686aade27285d6860b04689032d63189ee42676318b21556140421fL502)

Just wanted to reduce unnecessary diffs

[the remark this answers is not in the document]
Comment (sofiagvaladze) [reply addons/hr_holidays_attendance/models/hr_attendance.py:12]: Indeed, I missed that ^^

**Comment by sofiagvaladze** [comment]
@robodoo r+
@fw-bot up to saas-15.2

**Comment by sofiagvaladze** [comment]
@robodoo rebase-ff

**Pull request closed without merging**